lua/gatekeeper/staticlib.lua

```diff
@@ -124,6 +124,41 @@ function convert_numa_table_to_array (numa_table)
 	return res
 end
 
+function gk_sol_map (gk_lcores, sol_lcores)
+	local m = {}
+	local d = math.ceil(#gk_lcores / #sol_lcores)
+	local sol_remaining_cap = {}
+
+	if #gk_lcores % #sol_lcores ~= 0 then
+		print("Wrong GK-to-SOL assignment");
+	end
+
+	for i, v in ipairs(sol_lcores) do
+		table.insert(sol_remaining_cap, d)
```

Comment:
Notice that the code won't work for the case: 3 GK blocks and 2 NUMA nodes. Variable `d` is 1 in this case and only two entries are added to `sol_remaining_cap`. Although this case is unbalanced and already reported above in the code, we should support it anyway. A solution to address this issue goes as follows:

1. Rename `sol_remaining_cap` to `sol_allocated`;
2. Add zero instead of `d` to `sol_allocated`;
3. Discard variable `d`;
4. Instead of searching for the largest value in `sol_allocated` (see current test `sol_remaining_cap[i2] > sol_remaining[idx` later in the code), search for smallest;
5. Incrementing `sol_allocated` instead of decrementing it when a GK block is associated to a SOL block.